src/renderer/component/userFieldNew/view.jsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import { Form, FormRow, Submit } from 'component/form.js';
+
+class UserFieldNew extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      phone: '',
+      email: '',
+      country_code: '+1',
+    };
+
+    this.formatPhone = this.formatPhone.bind(this);
+  }
+
+  formatPhone(value) {
+    const { country_code } = this.state;
+    value = value.replace(/\D/g, '');
+    if (country_code === '+1') {
+      if (!value) {
+        return '';
+      } else if (value.length < 4) {
+        return value;
+      } else if (value.length < 7) {
+        return `(${value.substring(0, 3)}) ${value.substring(3)}`;
+      }
+      const fullNumber = `(${value.substring(0, 3)}) ${value.substring(3, 6)}-${value.substring(
+        6
+      )}`;
+      return fullNumber.length <= 14 ? fullNumber : fullNumber.substring(0, 14);
+    }
+    return value;
+  }
+
+  formatCountryCode(value) {
+    if (value) {
+      return `+${value.replace(/\D/g, '')}`;
+    }
+    return '+1';
+  }
+
+  handleChanged(event, fieldType) {
+    const formatter = {
+      email: _ => _,
+      phone: this.formatPhone,
+      country_code: this.formatCountryCode,
+    };
+    this.setState({
+      [fieldType]: formatter[fieldType](event.target.value),
+    });
+  }
+
+  handleSubmit() {
+    const { email, phone, country_code } = this.state;
+    if (phone) {
+      this.props.addUserPhone(phone.replace(/\D/g, ''), country_code.replace(/\D/g, ''));
+    } else {
+      this.props.addUserEmail(email);
+    }
+  }
+
+  render() {
+    const { cancelButton, emailErrorMessage, phoneErrorMessage, isPending, fieldType } = this.props;
+
+    return fieldType === 'phone' ? (
+      <div>
+        <p>
+          {__(
+            'Enter your phone number and we will send you a verification code. We will not share your phone number with third parties.'
+          )}
+        </p>
+        <Form onSubmit={this.handleSubmit.bind(this)}>
+          <FormRow
```

Comment:
Does this have to be collected separately as opposed to letting them type a full number? If it is separate field, could it at least be an inline select next to the full number entry?

Reply:
I suppose it could be a select. Twilio wants a separate country code. I tried to make this as fool proof and intuitive as possible. I think the way it is makes a lot of sense (it was based on a discussion in tech-app), but I am completely open to changing it.

Reply:
@liamcardenas yes, please make the country code a select displayed before the phone number entry.